vnext/Microsoft.ReactNative.SharedManaged/ReactMethodInfo.cs

```diff
@@ -2,304 +2,204 @@
 // Licensed under the MIT License.
 
 using Microsoft.ReactNative.Bridge;
-using Microsoft.ReactNative.Managed.Linq;
 using System;
-using System.Collections.Generic;
-using System.Linq.Expressions;
 using System.Reflection;
 using System.Threading;
-using ReflectionMethodInfo = System.Reflection.MethodInfo;
+using static Microsoft.ReactNative.Managed.JSValueGenerator;
+using static Microsoft.ReactNative.Managed.JSValueReaderGenerator;
+using static Microsoft.ReactNative.Managed.JSValueWriterGenerator;
+using static System.Linq.Expressions.Expression;
 
 namespace Microsoft.ReactNative.Managed
 {
   class ReactMethodInfo
   {
-    public ReactMethodInfo(ReflectionMethodInfo methodInfo, ReactMethodAttribute methodAttribute)
+    public ReactMethodInfo(MethodInfo methodInfo, ReactMethodAttribute methodAttribute)
     {
       MethodName = methodAttribute.MethodName ?? methodInfo.Name;
 
       Type returnType = methodInfo.ReturnType;
       ParameterInfo[] parameters = methodInfo.GetParameters();
-      if (returnType != typeof(void))
-      {
-        MethodReturnType = MethodReturnType.Callback;
-        MethodImpl = new Lazy<ReactMethodImpl>(() => MakeReturningMethod(methodInfo, returnType, parameters), LazyThreadSafetyMode.PublicationOnly);
-      }
-      else
+      Type lastParameterType = parameters.Length > 0 ? parameters[parameters.Length - 1].ParameterType : null;
+      Type secondToLastParameterType = parameters.Length > 1 ? parameters[parameters.Length - 2].ParameterType : null;
+      Func<ReactMethodImpl> createMethod;
+
+      bool isPromise(Type type)
```

Comment:
>isPromise [](start = 11, length = 9)

Would this be cleaner as a private method on the class instead of a method inside of a method?

Reply:
Maybe. I can revisit it later.

---
In reply to: [356823803](https://github.com/microsoft/react-native-windows/pull/3760#discussion_r356823803) [](ancestors = 356823803)